.github/workflows/c-chain-benchmark.yml

```diff
@@ -0,0 +1,77 @@
+        required: false
+        default: 's3://avalanchego-bootstrap-testing/cchain-mainnet-blocks-1m-ldb.zip'
+      current-state-dir:
+        description: 'The current state directory. Supports S3 directory/zip and local directories.'
+        required: false
+        default: 's3://avalanchego-bootstrap-testing/cchain-current-state-hashdb-100.zip'
+
+jobs:
+    c-chain-reexecution:
+      permissions:
+        id-token: write
+        contents: write
+      runs-on: ubuntu-latest
+      steps:
+        - name: Configure AWS Credentials
+          uses: aws-actions/configure-aws-credentials@v4
+          with:
+            role-to-assume: ${{ secrets.AWS_S3_READ_ONLY_ROLE }}
+            aws-region: us-east-2
+        - name: Set task env via GITHUB_ENV
+          id: set-params
+          run: |
+            if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+              echo "START_BLOCK=${{ github.event.inputs.start-block }}"
+              echo "END_BLOCK=${{ github.event.inputs.end-block }}"
+              echo "SOURCE_BLOCK_DIR=${{ github.event.inputs.source-block-dir }}"
+              echo "CURRENT_STATE_DIR=${{ github.event.inputs.current-state-dir }}"
+            else
+              echo "START_BLOCK=100"
+              echo "END_BLOCK=10000"
+              echo "SOURCE_BLOCK_DIR=s3://avalanchego-bootstrap-testing/cchain-mainnet-blocks-1m-ldb.zip"
+              echo "CURRENT_STATE_DIR=s3://avalanchego-bootstrap-testing/cchain-current-state-hashdb-100.zip"
+            fi
+        - uses: actions/checkout@v4
+        - uses: ./.github/actions/setup-go-for-project
+        - name: Run C-Chain Re-Execution
+          uses: ./.github/actions/run-monitored-tmpnet-cmd
+          with:
+            run: ./scripts/run_task.sh reexecute-cchain-range-with-copied-data EXECUTION_DATA_DIR=$GITHUB_WORKSPACE/reexecution-data
+            artifact_prefix: c-chain-reexecute-range
+            prometheus_username: ${{ secrets.PROMETHEUS_ID || '' }}
+            prometheus_password: ${{ secrets.PROMETHEUS_PASSWORD || '' }}
+            grafana_dashboard_id: 'c-chain-processing-custom-v8/c-chain-processing'
+            loki_username: ${{ secrets.LOKI_ID || '' }}
+            loki_password: ${{ secrets.LOKI_PASSWORD || '' }}
+            runtime: "" # Set runtime input to empty string to disable log collection
+        - name: Download Previous Benchmark Result
+          uses: actions/cache@v4
+          with:
+            path: ./cache
+            key: ${{ runner.os }}-reexecute-cchain-range-benchmark.json
+        - name: Compare Benchmark Results
+          uses: benchmark-action/github-action-benchmark@v1
+          with:
+            tool: 'go'
+            output-file-path: $GITHUB_WORKSPACE/reexecution-data/reexecute-cchain-range.txt
+            external-data-json-path: ./cache/${{ runner.os }}-reexecute-cchain-range-benchmark.json
```

Comment:
Writing to the cache and comparing against the last entry enables a comparison.

This should be changed to write to the cache only on the master branch for comparison and other triggers should compare against the latest baseline set by master